### ISSUE TITLE
[Feature Request] Subject Signal

### ISSUE THREAD

**Issue by hillin** [angular/angular#58863]
### Which @angular/* package(s) are relevant/related to the feature request?

core

### Description

I find myself constantly running into the situation that for a given reactive value, I need to use it both in rx and signal contexts. 

For instance let's say there is an `selectedItem` value, representing an entity selected by the user. We would like to store it in a `WritableSignal` so something could be derived out of it with `computed`; we also want to use it to retrieve some related data from the backend, thus it would also be great to be an observable (to be `switchMap`-ed).

In the meantime we just make the value a signal and spam `toObservable`s whenever we need it in rx. There are so many of them that I really considered it as pattern that should be addressed with better ergonomics.

### Proposed solution

We came up with a solution like this:
```typescript
import { Signal } from "@angular/core";
import { toSignal } from "@angular/core/rxjs-interop";
import { BehaviorSubject, Observable } from "rxjs";

/** A signal backed by a subject. */
export interface SubjectSignal<T> extends Signal<T> {
  observable: Observable<T>;
  update(updateFn: (value: T) => T): void;
  set(value: T): void;
}

/** Make a signal that is backed by a subject. */
export function subjectSignal<T>(initialValue: T): SubjectSignal<T> {
  const subject = new BehaviorSubject<T>(initialValue);
  const signal = toSignal(subject) as SubjectSignal<T>;

  signal.observable = subject.asObservable();
  signal.update = (updateFn: (value: T) => T) => {
    subject.next(updateFn(subject.value));
  };
  signal.set = (value: T) => {
    subject.next(value);
  };

  Object.freeze(signal);

  return signal;
}
```

So it's basically a wrapper to `toSignal`, but an observable could be retrieved out from it.

### Alternatives considered

Ideally we could have `SubjectSignal<T>` to be used both as a signal and as an observable:
```typescript
interface SubjectSignal<T> extends Observable<T>, Signal<T>;
```
but I'm not sure if it's possible, because `Signal<T>` has a callable signature and `Observable<T>` is a class. Anyways we would like to see a more elegant solution.

**Comment by e-oz** [comment]
1. Check this out: https://ngxtension.netlify.app/utilities/signals/to-observable-signal/
2. Providing it out of the box by the framework would be awesome (with on-demand-only creation of observables).

**Comment by pkozlowski-opensource** [comment]
While this is true that many practical situations today require toSignal / toObservable conversions, we expect that those cases will diminish with time. As an example, the newly introduced `resource` makes it possible to retrieve data from the backend.

We were toying with a migration construct that would be a mix of a signal / observable, but the current thinking is that it might be more confusing than helpful. 

I will keep this issue open so we can capture interest / duplicates, but it is very unlikely that we will introduce such APi in the near future.

**Comment by hillin** [comment]
I agree that the `resource` construct looks very promising and could address most of the said use cases on our end.

**Comment by JeanMeche** [comment]
More than a year later, we didn't notice any traction on the topic. I'll close this.
